examples/app.py

```diff
@@ -0,0 +1,116 @@
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+
+"""Minimal Flask application example for development.
+
+Create database and tables:
+
+.. code-block:: console
+
+   $ cd examples
+   $ flask -a app.py db init
+   $ flask -a app.py db create
+
+Create some users:
+
+.. code-block:: console
+   $ flask -a app.py accounts users create -e info@invenio-software.org -a
+   $ flask -a app.py accounts users create -e editor@invenio-software.org -a
+   $ flask -a app.py accounts users create -e admin@invenio-software.org -a
+
+Add a role to a user:
+   $ flask -a app.py accounts roles create -n admin
+   $ flask -a app.py accounts roles add -u admin@invenio-software.org -r admin
+
+Assign some allowed actions to this users:
+
+.. code-block:: console
+   $ flask -a app.py access users allow -a open -e editor@invenio-software.org
+   $ flask -a app.py access roles allow -a open -r admin
+
+Run the development server:
+
+   $ flask -a app.py --debug run --debugger
+   $ flask -a app.py shell
+
+"""
+
+from __future__ import absolute_import, print_function
+
+from flask import Flask, render_template
+from flask_babelex import Babel
+
+from flask_cli import FlaskCLI
+from flask_mail import Mail
+from invenio_access import InvenioAccess
+from invenio_db import InvenioDB
+from invenio_accounts import InvenioAccounts
+from flask_security import current_user
+from flask.ext.principal import Permission, RoleNeed, ActionNeed
+
+# Create Flask application
+app = Flask(__name__)
+app.secret_key = 'ExampleApp'
+FlaskCLI(app)
+Babel(app)
+Mail(app)
+InvenioDB(app)
+InvenioAccounts(app)
+InvenioAccess(app)
+
+
+@app.route("/")
+def index():
+    """Basic test view."""
+    if current_user.is_anonymous():
+        return render_template("invenio_access/open.html")
+    else:
+        message = "Hi %s" % current_user.email
+        return render_template("invenio_access/limited.html", message=message)
+
+
+admin_permission = Permission(RoleNeed('admin'))
+
+
+@app.route('/role_admin')
+@admin_permission.require()
+def role_admin():
+    """View only allowed to admin role."""
+    message = 'You are opening a page limited to admin role.'
+    return render_template("invenio_access/limited.html", message=message)
+
+
+open_permission = Permission(ActionNeed('open'))
+
+
+@app.route('/action_open')
+@open_permission.require()
+def action_open():
+    """View only allowed to open action."""
+    message = 'You are opening a page limited to action open'
+    return render_template("invenio_access/limited.html", message=message)
+
+if __name__ == "__main__":
```

Comment:
you can remove these last three lines as one should run `flask --app app run` instead.